src/python/pants/backend/awslambda/python/rules.py

```diff
@@ -54,6 +55,17 @@ class PythonAwsLambdaFieldSet(PackageFieldSet):
 async def package_python_awslambda(
     field_set: PythonAwsLambdaFieldSet, lambdex: Lambdex, union_membership: UnionMembership
 ) -> BuiltPackage:
+
+    if Platform.is_macos:
+        logger.warning(
```

Comment:
This might be useful information if the packing fails, but it's otherwise continual noise every time this is run and packaging succeeds. How about switching to a FallibleProcessResult and then conditioning this warning on both macOS and process failure?

Reply:
Agreed that would be a superior experience.

Per discussion at https://github.com/pantsbuild/pants/issues/11941, implementing it is trickier. We need to 

1. Add a `FalliblePex` type
2. Allow augmenting a `ProcessExecutionFailure` with additional details.
     - I started on this last week as part of https://github.com/pantsbuild/pants/issues/13136, but Benjy and I decided to punt on the ticket and the design work of how to show the extra info. 

So I suggested that using a warning is a good first start to get some progress.

Reply:
Erm. Ok. I'm not a fan of bridging the gap with noise, but looks like others are.

Reply:
Opened https://github.com/pantsbuild/pants/issues/13865 to track that second task.